### PR TITLE
Improve artifact upload

### PR DESCRIPTION
**Pull request by berryd**
### Description
QMR differs in many ways from the other apps, so some of the issues with artifact naming convention collisions won't exit, but there is still some minor opportunity for improvement, and in the course of this work, I will verify that the uploads do in fact work.  

One thing specifically different in QMR than other applications is that e2e tests run concurrently in a GHA matrix.  This produces an interesting problem of a race condition being able to override a previously saved artifact should more than one runner fail.  This problem is mitigated by prefixing the artifact with the name of the matrix (the matrix is a singleton because it determines the path to the test to execute)

### Related ticket(s)

CMDCT-3556

---
### How to test

See [this run](https://github.com/Enterprise-CMCS/macpro-mdct-qmr/actions/runs/8882953583) for an example.  For this instance, I forced two of the e2e tests to fail to demonstrate that the artifacts could be successfully saved.  A successful run will produce no artifacts.

### Notes

---
### Pre-review checklist

- [ ] I have added [thorough](https://shorturl.at/aejkF) tests, if necessary
- [ ] I have updated relevant documentation, if necessary
- [ ] I have performed a self-review of my code
- [ ] I have manually tested this PR in the deployed cloud environment

---
#### Review
- [ ] Design: This work has been reviewed and approved by design, if necessary
- [ ] Product: This work has been reviewed and approved by product owner, if necessary

#### Security
_If either of the following are true, notify the team's ISSO (Information System Security Officer)._

- [ ] These changes are significant enough to require an update to the SIA.
- [ ] These changes are significant enough to require a penetration test.
---

_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_

### CODE REVIEW

File: .github/workflows/cypress-workflow.yml [hunk 1]
```diff
@@ -69,9 +69,13 @@ jobs:
           CYPRESS_STATE_USER_4: ${{ secrets.cypress-user4 }}
           CYPRESS_ADMIN_USER: ${{ secrets.cypress-admin-user }}
           CYPRESS_QMR_PASSWORD: ${{ secrets.cypress-password }}
+      - name: How can I make this unique
+        run: |
+          cat $GITHUB_ENV
```
Comment (gmrabian) [inline .github/workflows/cypress-workflow.yml:74]: meant to stay?

**Review by gmrabian — APPROVED** [review]
good except for one lingering "debug" step (?)

File: .github/workflows/cypress-workflow.yml (hunk 1 above)
Comment (berryd) [reply .github/workflows/cypress-workflow.yml:74]: nope

Comment (berryd) [reply .github/workflows/cypress-workflow.yml]: Cleanup pushed

**Review by gmrabian — APPROVED** [review]

**Pull request merged**